### DISCUSSION TITLE
Android使用PAGPlayer和PAGSurface导出视频很慢

### DISCUSSION THREAD

**Discussion by mozushixin** [Tencent/libpag#2369] [General] [answered]
如题
场景:PAG模板中有750帧，时长30s
用MediaCodec创建离屏surface编码，然后muxer合成视频
但是pagPlayer flush方法每帧耗时在100ms左右，所以导致合成视频的时长很久
请问有优化的建议吗？

**Comment by domchen** [comment, marked as answer]
这种问题一般是优化你的素材。检查里面的大面积的BMP预合成，避免使用过多的BMP预合成或者不用BMP预合成性能会有立竿见影的变化。但是不要欺骗自己的方式把本来的BMP预合成拆成无所个图片图层一个个轮播，这个是比BMP预合成性能更差的方式。
